test_unstructured/partition/html/test_parser.py

```diff
@@ -89,6 +85,263 @@ def test_normalize_text_produces_normalized_text(text: str, expected_value: str)
     assert _normalize_text(text) == expected_value
 
 
+# -- PHRASING ACCUMULATORS -----------------------------------------------------------------------
+
+
+class Describe_PhraseAccumulator:
+    """Isolated unit-test suite for `unstructured.partition.html.parser._PhraseAccumulator`."""
+
+    def it_is_empty_on_construction(self):
+        accum = _PhraseAccumulator()
+
+        phrase_iter = accum.flush()
+
+        with pytest.raises(StopIteration):
+            next(phrase_iter)
+
+    # -- .add() -----------------------------------------------------------
+
+    def it_accumulates_text_segments(self):
+        accum = _PhraseAccumulator()
+
+        accum.add(TextSegment("Ford... you're turning ", {}))
+        accum.add(TextSegment("into a penguin.", {}))
```

Comment:
🐧

Reply:
https://www.goodreads.com/quotes/4388-ford-you-re-turning-into-a-penguin-stop-it :)